src/ast/nodes/SequenceExpression.ts

```diff
@@ -16,54 +39,84 @@ export default class SequenceExpression extends NodeBase {
 		return this.expressions.some(expression => expression.hasEffects(options));
 	}
 
+	hasEffectsWhenAccessedAtPath(path: ObjectPath, options: ExecutionPathOptions): boolean {
+		return (
+			path.length > 0 &&
+			this.expressions[this.expressions.length - 1].hasEffectsWhenAccessedAtPath(path, options)
+		);
+	}
+
+	hasEffectsWhenAssignedAtPath(path: ObjectPath, options: ExecutionPathOptions): boolean {
+		return (
+			path.length === 0 ||
+			this.expressions[this.expressions.length - 1].hasEffectsWhenAssignedAtPath(path, options)
+		);
+	}
+
+	hasEffectsWhenCalledAtPath(
+		path: ObjectPath,
+		callOptions: CallOptions,
+		options: ExecutionPathOptions
+	): boolean {
+		return this.expressions[this.expressions.length - 1].hasEffectsWhenCalledAtPath(
+			path,
+			callOptions,
+			options
+		);
+	}
+
 	includeInBundle() {
 		let addedNewNodes = !this.included;
 		this.included = true;
-		if (this.expressions[this.expressions.length - 1].includeInBundle()) {
-			addedNewNodes = true;
+		for (let i = 0; i < this.expressions.length - 1; i++) {
+			const node = this.expressions[i];
+			if (node.shouldBeIncluded() && node.includeInBundle()) addedNewNodes = true;
 		}
-		this.expressions.forEach(node => {
-			if (node.shouldBeIncluded()) {
-				if (node.includeInBundle()) {
-					addedNewNodes = true;
-				}
-			}
-		});
+		if (this.expressions[this.expressions.length - 1].includeInBundle()) addedNewNodes = true;
 		return addedNewNodes;
 	}
 
-	render(code: MagicString, options: RenderOptions) {
+	reassignPath(path: ObjectPath, options: ExecutionPathOptions) {
+		if (path.length > 0) this.expressions[this.expressions.length - 1].reassignPath(path, options);
+	}
+
+	render(
+		code: MagicString,
+		options: RenderOptions,
+		{ renderedParent, fieldOfRenderedParent }: NodeRenderOptions = BLANK
+	) {
 		if (!this.module.graph.treeshake) {
 			super.render(code, options);
 		} else {
-			const last = this.expressions[this.expressions.length - 1];
-			last.render(code, options);
-
-			if (
-				this.parent.type === NodeType.CallExpression &&
-				last.type === NodeType.MemberExpression &&
-				this.expressions.length > 1
-			) {
-				this.expressions[0].included = true;
-			}
-
-			const included = this.expressions
-				.slice(0, this.expressions.length - 1)
-				.filter(expression => expression.included);
-			if (included.length === 0) {
-				code.remove(this.start, last.start);
-				code.remove(last.end, this.end);
-			} else {
-				let previousEnd = this.start;
-				for (const expression of included) {
-					expression.render(code, options);
-					code.remove(previousEnd, expression.start);
-					code.appendLeft(expression.end, ', ');
-					previousEnd = expression.end;
+			let firstStart = 0,
+				lastEnd,
+				includedNodes = 0;
+			for (const { node, start, end } of getCommaSeparatedNodesWithBoundaries(
```

Comment:
Nice!

Reply:
When I wrote `getCommaSeparatedNodesWithBoundaries`, I always hoped I could use it not only for variable declarations but also for sequence expressions. So should you decide to write your entire app using sequence expressions, you now get really nice comment handling 😉